examples/agents/in-house-ctf/dataset.py

```diff
@@ -0,0 +1,50 @@
+import os
+from glob import glob
+
+from inspect_ai.dataset import Dataset, Sample, json_dataset
+
+DATA_DIR = "data"
+SOLUTION_SIGNATURE = "## Possible solution"
+
+
+def read_dataset(config_name: str, shuffle: bool=False) -> Dataset:
+    # record to sample
+    def record_to_sample(record):
+        # collect files
+        task_id = record["challenge_name"]
+        task_assets = f"{DATA_DIR}/{task_id}"
+        task_files = [
+            os.path.relpath(f, task_assets)
+            for f in glob(os.path.join(task_assets, '**'), recursive=True)
+            if os.path.isfile(f)
+        ]
+        files = {os.path.basename(file): file for file in task_files}
+
+        # read solution
+        with open(f"{task_assets}/README.md") as f:
+            readme = f.read()
+
+        solution = ""
+        if SOLUTION_SIGNATURE in readme:
+            solution = readme.split(SOLUTION_SIGNATURE)[1].strip("\n")
+
+        return Sample(
+            id=task_id,
+            input=record["prompts"],
+            metadata={
+                "challenge_address": record["challenge_address"],
+                "objective_prompt": record["prompts"],
+                "solution": solution,
+                "get_flag_cmd": record["get_flag_cmd"],
+            },
+            files=files,
+            sandbox=("docker", f"{task_assets}/compose.yaml"),
+        )
+
+    excluded = []
```

Comment:
I just added this in case anyone wants to test things out and don't need run the full task. You're right I should make this as an input param for the `read_dataset()` function